lib/flexirest/configuration.rb

```diff
@@ -135,6 +135,25 @@ def password=(value)
         @@password = value
       end
 
+      DEFAULT_BASIC_URL_METHOD = :url
+      VALID_BASIC_URL_METHODS = [:url, :header]
+
+      def basic_auth_method(*params)
+        case params.size
+        when 0 # Used as a getter method
+          @basic_auth_method || DEFAULT_BASIC_URL_METHOD
+        when 1 # Used as a setter method
+          method_type = params.first
+          unless [:header, :url].include?(method_type)
+            raise %(Invalid basic_auth_method #{method_type.inspect}. Valid methods are #{VALID_BASIC_URL_METHODS.inspect}.)
+          end
+
+          @basic_auth_method = method_type
+        else
+          raise "basic_auth_method can only take one parameter"
+        end
+      end
```

Comment:
I would replace this with a simple `def basic_auth_method(method = nil)` and then check for the value being nil (and if so return it), if not, set it. The same pattern we use elsewhere.

Reply:
You are right, that is much simpler!